### ISSUE TITLE
`torch.cuda.amp.autocast` in animeinsseg annotator + dead `torch.cuda.amp` import in loss_depth.py (deprecated since torch 2.4)

### ISSUE THREAD

**Issue by xyf5432** [shitagaki-lab/see-through#36]
## Summary

Three real uses of the deprecated `torch.cuda.amp.autocast` in the annotator path, plus a near-dead `torch.cuda.amp` import in the training code (see below). `torch.cuda.amp.autocast` has been deprecated since torch 2.4 (and `GradScaler` since 2.3); both are scheduled for removal.

1. [annotators/animeinsseg/skytnt_segmentation.py:30](https://github.com/shitagaki-lab/see-through/blob/main/annotators/animeinsseg/skytnt_segmentation.py#L30):
   ```python
   with torch.cuda.amp.autocast():
   ```
   (inference, reached via `inference/scripts/parse_live2d.py` and the UI thread)

2. [annotators/animeinsseg/rtmdet_inshead_custom.py:156](https://github.com/shitagaki-lab/see-through/blob/main/annotators/animeinsseg/rtmdet_inshead_custom.py#L156) and [:275](https://github.com/shitagaki-lab/see-through/blob/main/annotators/animeinsseg/rtmdet_inshead_custom.py#L275):
   ```python
   with torch.cuda.amp.autocast(enabled=False):
   ```
   (this file is the repo's own "custom" fork of the RTMDet head, not pristine vendored code)

3. [training/train/loss_depth.py:28](https://github.com/shitagaki-lab/see-through/blob/main/training/train/loss_depth.py#L28):
   ```python
   import torch.cuda.amp as amp
   ```
   Nearly a dead import — the only remaining reference is the commented-out block at line 166 (`# with amp.autocast(enabled=False):  # amp causes NaNs in this loss function`). A plain `import torch.cuda.amp` emits no warning on torch 2.4+ (verified on torch 2.11; only *calling* `autocast`/`GradScaler` warns), so this line is not itself a warning source. Since the import exists solely for that commented-out block, we suggest **not deleting it here** — the block and the import should be removed together by the maintainers if the block is deemed obsolete. For this PR we keep the import but retarget it to `torch.amp` (annotated), and update the commented call to the new API so the import carries no deprecated reference.

The pinned `torch==2.8.0+cu128` (aarch64: `>=2.9.0`) is squarely inside the deprecated range, so every annotator run emits the `FutureWarning`.

## How to reproduce

```bash
pip install torch==2.8.0  # as pinned in requirements.txt
python -c "
import warnings; warnings.simplefilter('error', FutureWarning)
with torch.cuda.amp.autocast(enabled=False):
    pass
"
```

```
FutureWarning: torch.cuda.amp.autocast is deprecated. Please use torch.amp.autocast('cuda', ...)
```

(Verified on torch 2.11: the warning fires on every `autocast` call.)

## Request

Migrate the three uses to `torch.amp.autocast("cuda", ...)` (available since torch 2.0, well below the 2.8.0 floor). In `loss_depth.py`, keep the import but retarget it to `torch.amp` and annotate it (removal is left to the maintainers, together with the commented-out block it serves). Keep the `enabled=False` semantics at lines 156/275 — the equivalent is `torch.amp.autocast("cuda", enabled=False)`.

## Suggested fix

```diff
 # annotators/animeinsseg/skytnt_segmentation.py:30
-            with torch.cuda.amp.autocast():
+            with torch.amp.autocast("cuda"):
 # annotators/animeinsseg/rtmdet_inshead_custom.py:156, 275
-            with torch.cuda.amp.autocast(enabled=False):
+            with torch.amp.autocast("cuda", enabled=False):
 # training/train/loss_depth.py:28 — keep the import, retarget it to torch.amp
-import torch.cuda.amp as amp
+# torch.cuda.amp is deprecated since torch 2.4; the import below targets
+# torch.amp (same API, no warning). Kept only for the commented-out block
+# below; if that block is removed, this import should go with it.
+import torch.amp as amp
 # training/train/loss_depth.py:166 — the commented call, updated for the new API
-#         with amp.autocast(enabled=False):  # amp causes NaNs in this loss function
+#         with amp.autocast("cuda", enabled=False):  # amp causes NaNs in this loss function
```

## References

- PyTorch deprecation: `torch.cuda.amp.autocast` deprecated since 2.4; `torch.amp` is the replacement namespace.
- Same migration precedent: [ultralytics/yolov5#13244](https://github.com/ultralytics/yolov5/pull/13244), [huggingface/lerobot#3167](https://github.com/huggingface/lerobot/pull/3167).
